app/serializers/current_user_serializer.rb

```diff
@@ -309,4 +311,14 @@ def redesigned_topic_timeline_enabled
       false
     end
   end
+  def custom_sidebar_sections_enabled
+    if SiteSetting.enable_custom_sidebar_sections.present?
+      GroupUser.exists?(
+        user_id: object.id,
+        group_id: SiteSetting.enable_custom_sidebar_sections.split("|"),
+      )
+    else
+      false
+    end
```

Comment:
```suggestion
  def custom_sidebar_sections_enabled
    if SiteSetting.enable_custom_sidebar_sections.present?
      object.in_any_groups?(SiteSetting.enable_custom_sidebar_sections_map)
    else
      false
    end
```

We have a method to get the group ids from site settings of type `:group_list`.

https://github.com/discourse/discourse/blob/9d55e2a9394cfcdac3041ae75cc34a6ea417be43/lib/site_setting_extension.rb#L557-L564